src/scripts/party/evolutions/EvolutionHandler.ts

```diff
@@ -55,10 +56,11 @@ class EvolutionHandler {
                 evolvedPartyPokemon.heldItem = basePartyPokemon.heldItem;
             }
             App.game.party.removePokemonByName(data.basePokemon);
+            baseRemoved = true;
         }
 
         // EVs
         evolvedPartyPokemon.effortPoints += App.game.party.calculateEffortPoints(evolvedPartyPokemon, shiny, GameConstants.STONE_EP_YIELD);
-        return shiny;
+        return shiny || baseRemoved;
```

Comment:
I don't like this. The callers of this function can no longer know if a `true` response means shiny.
Instead, I think you should be returning a more detailed value like
```ts
return {
    shiny,
    baseRemoved,
}
```
and update anywhere calling this function to handle the response however they need to.
Additionally the `return false` guards at the top can probably just return `null` to signal no evolution was done.

I would probably pass this return type out of PartyPokemon.useStone and EvolutionItem.use as isdirectly, and handle the logic up in ItemHandler.useStones

Reply:
Would it look better if I just test if the mon is still in the party in [ItemHandler](https://github.com/pokeclicker/pokeclicker/blob/develop/src/modules/items/ItemHandler.ts#L87), then ? So everything else remains untouched and preserved.

Reply:
Yeah that would be fine, getPokemon and getPokemonByName both use a Map lookup so it isn't an expensive operation to be doing